From a6aaecd4a06429f92b99895da0d7dba40a158f31 Mon Sep 17 00:00:00 2001
From: Sailaja <sbandarupalli@habana.ai>
Date: Tue, 09 Mar 2021 10:40:18 +0200
Subject: [PATCH] [SW-13692] hl: fix thermal function calls

Adding the fix for the calibration, which utilized efuse values read
this was fixed in u-boot but not in preboot

Signed-off-by: Sailaja <sbandarupalli@habana.ai>
Change-Id: Ie0e9fef1bcbf9c7810f541c7618eb63df2fa7ad7
---

diff --git a/preboot/thermal.c b/preboot/thermal.c
index 5dca696..296f8ed 100644
--- a/preboot/thermal.c
+++ b/preboot/thermal.c
@@ -232,10 +232,10 @@
 			preboot_fuse_read(hl_ts_data[ts].calib_offset,
 						&efuse_word);
 
-			/* Calculate new A and B coeff using efuse offsets */
-			hl_tmp_extract_A_B(efuse_word, ts);
 			hl_set_calib_coeff(ts, true,
 						&hl_ts_data[ts].hl_tmp_data);
+			/* Calculate new A and B coeff using efuse offsets */
+			hl_tmp_extract_A_B(efuse_word, ts);
 		}
 
 		hl_ts_alarm_set(ts);
